lib/spec/swagger/utils.js

```diff
@@ -266,9 +266,29 @@ function prepareSwaggerMethod (schema, ref, swaggerObject) {
   return swaggerMethod
 }
 
+function prepareSwaggerDefinitions (definitions, ref) {
+  return Object.entries(definitions)
+    .reduce((res, [name, definition]) => {
+      const _ = { ...definition }
+      const resolved = ref.resolve(_, { externalSchemas: [definitions] })
+
+      // Swagger doesn't accept $id on /definitions schemas.
+      // The $ids are needed by Ref() to check the URI so we need
+      // to remove them at the end of the process
+      delete resolved.$id
+      delete resolved.definitions
+
+      return {
```

Comment:
I do not catch it in first place. But we should prevent a `reduce` loop and `spread` operator in here.
It create a new object each round of the loop.

```js
res[name] = resolved
return res
```

The above code should be enough.

Reply:
Benchmark

```js
const { performance } = require('perf_hooks');
const arr = Array(10000).fill(0)

function speedSpread() {
  const start = performance.now()
  const obj = arr.reduce(function(obj) {
    return { ...obj, [Math.random()]: Math.random() } 
  }, {})
  return performance.now() - start
}

function speedPropForEach() {
  const start = performance.now()
  const obj = {}
  arr.forEach(function() {
    obj[Math.random()] = Math.random()
  })
  return performance.now() - start
}

function speedPropReduce() {
  const start = performance.now()
  const obj = arr.reduce(function(obj) {
    obj[Math.random()] = Math.random()
    return obj
  }, {})
  return performance.now() - start
}

const a = speedSpread()
const b = speedPropForEach()
const c = speedPropReduce()

console.log('speedSpread', a)
console.log('speedPropForEach', b)
console.log('speedPropReduce', c)
```

Reply:
nice - updated for openAPI and swagger 👍